example/generator/odp_generator.c

```diff
@@ -379,6 +395,29 @@ static int setup_udp_pkt(odp_packet_t pkt, odp_pktout_config_opt_t *pktout_cfg,
 		ip->chksum = ~odp_chksum_ones_comp16(ip, ODPH_IPV4HDR_LEN);
 	}
 
+	if ((udp_arg->srcport_start != udp_arg->srcport_end) ||
+	    (udp_arg->dstport_start != udp_arg->dstport_end)) {
+		odph_udphdr_t *udp = (odph_udphdr_t *)(buf + ODPH_ETHHDR_LEN +
+						       ODPH_IPV4HDR_LEN);
+
+		if (udp_arg->srcport_start != udp_arg->srcport_end) {
+			udp->src_port = odp_cpu_to_be_16(udp_arg->srcport_crt);
+			if (udp_arg->srcport_crt >= udp_arg->srcport_end)
+				udp_arg->srcport_crt = udp_arg->srcport_start;
+			else
+				udp_arg->srcport_crt++;
```

Comment:
So do all threads inc the single port counter variable? I think for performance reasons there has to be counter per thread  instead of sharing global variable.

Reply:
It is one per thread already. See udp_param = &args->thread[i].tx.udp_param;

Reply:
Application is keeping a set of per thread variables:

typedef struct {
.................
	/** Thread specific arguments */
	thread_args_t thread[MAX_WORKERS];
................
} args_t;

typedef struct {
.................
      union {
		struct {
......................
			udp_args_t udp_param;  /**< UDP configuration */
		} tx;
............................
	};
.................
} thread_args_t;

So, port counters are kept per thread.